add GenericMemoryRef address space regression test

fjralf opened the pull request:
Add test case for LLVM IR generation bug where stores to GenericMemoryRef incorrectly use addrspace(10) instead of addrspace(11) in bounds checking code, causing assertion failures in LLVM optimization passes.

This reproduces the issue described in the gist where bounds checking generates incorrect address space casts in the LLVM IR.

note : 
- Tested on CI due to local environment limitations (Termux/Android)

issue #59510

fjralf committed:
test: add GenericMemoryRef address space regression test

Add test case for LLVM IR generation bug where stores to GenericMemoryRef
incorrectly use addrspace(10) instead of addrspace(11) in bounds checking
code, causing assertion failures in LLVM optimization passes.

This reproduces the issue described in the gist where bounds checking
generates incorrect address space casts in the LLVM IR.

fjralf committed:
test: add GenericMemoryRef address space regression test

Add test case for LLVM IR generation bug where stores to GenericMemoryRef
incorrectly use addrspace(10) instead of addrspace(11) in bounds checking
code, causing assertion failures in LLVM optimization passes.

This reproduces the issue described in the gist where bounds checking
generates incorrect address space casts in the LLVM IR.

vtjnash commented on test/llvmpasses/memoryref-addrspace.jl:
```diff
@@ -0,0 +1,21 @@
+# CHECK-NOT: store {{.*}} addrspace(10) {{.*}}GenericMemoryRef
+# CHECK: call void @ijl_bounds_error_int
+
+emit(bf, Tuple{Int, Vector{Float64}})
```
```suggestion
emit(bf, Int, Vector{Float64})
```

vtjnash reviewed (commented):
SGTM. But notably is failing in CI due to incorrect call to `emit`.

fjralf committed:
Update emit call to use idiomatic argument syntax as suggested by vtjnash

rickwoi commented:
> SGTM. Namun, khususnya gagal di CI karena pemanggilan yang salah ke `emit`.

Fixed the emit call per your suggestion. CI should be green now.

fjralf committed:
test: fix GenericMemoryRef address space regression test pattern

Update CHECK pattern to match current LLVM IR output after address space bug fix.

fjralf committed:
Merge branch 'master' into add-memoryref-addrspace-test

fjralf committed:
Merge branch 'master' into add-memoryref-addrspace-test
